### PR TITLE
Improve test code quality rs

### PR DESCRIPTION
**Pull request by RinorSefa**
In the **datasetTest** class removed code smell of **assertion Roulette** and **eager test** by dividing/extracting the testDataset method into testDSnumFeatures,testDSnumFeatureTypes, testDSsize, testDSnumFeatureTokens, testDSnumClasses, testLinearClassifierProbabilityUsingDS, testLinearClassifierClassOfUsing DS.

In the **CategoryWordTagFactoryTest** first removed code smells of **eager Test** by extracting the method, and then removed code smell of **indirect testing** because it was testing another class method CategoryWordTagTest methods, which is tested in its class.

In the **CategoryWordTagTest** removed code smell of **eager Test** by dividing/extracting testCopy method to two methods test testCategoryWordTagConstructorWithExistingLabel and testCategoryWordTagConstructorWithParCategoryWordTag, and also removed **assertion roulette** by including test messages.

In the **DirectedMultiGraphTest** removed code smell of **eager test** by spliting method testRemove into 4 different methods, testRemoveEdges, testRemoveNonExistingEdges, testRemoveVertex, testRemoveNonExistingVertex.

In the **RulesTest** class, testIsAcronym method, removed the **assertion Roulette** test smell by including test messages.

In the **SpanishUnknownWordSignature** class, removed **assertion roulette** test smell by including test messages.

In the **SloppyMathTest** class method, testRoundWithBoundaries removed the **assertion roulette** test smell by including test messages.

### CODE REVIEW

**Comment by jvdassen** [comment]
Structure looks clean, tests seem well done!
I only have found some minor issues with formatting:

- `0fd33c`, `7f1d79`, `8a1e09` show some issues where there is sometimes a space missing between the arguments. E.g:
```
     assertEquals("Testing method category","A", tag2.category()); 
```
could be changed to:
```
     assertEquals("Testing method category", "A", tag2.category()); 
```
- There are four subsequent commit messages "` removed code smells of Assertion Roulette `". Maybe one could use something like "Remove Assertion Roulette from CategoryWordTagTest"? But this is just a minor thing I guess.

**Comment by RinorSefa** [comment]
> y

Thank you for feedback, I fixed the issues you found

**Comment by RinorSefa** [comment]
> 
> 
> Structure looks clean, tests seem well done!
> I only have found some minor issues with formatting:
> 
>     * `0fd33c`, `7f1d79`, `8a1e09` show some issues where there is sometimes a space missing between the arguments. E.g:
> 
> 
> ```
>      assertEquals("Testing method category","A", tag2.category()); 
> ```
> 
> could be changed to:
> 
> ```
>      assertEquals("Testing method category", "A", tag2.category()); 
> ```
> 
>     * There are four subsequent commit messages "`removed code smells of Assertion Roulette`". Maybe one could use something like "Remove Assertion Roulette from CategoryWordTagTest"? But this is just a minor thing I guess.

Thank you for your feedback i fixed the issues presented

File: test/src/edu/stanford/nlp/classify/DatasetTest.java [hunk 1]
```diff
@@ -1,53 +1,92 @@
...
-  public void testDataset() {
-    Dataset<String, String> data = new Dataset<>();
-    data.add(new BasicDatum<String, String>(Arrays.asList(new String[]{"fever", "cough", "congestion"}), "cold"));
-    data.add(new BasicDatum<String, String>(Arrays.asList(new String[]{"fever", "cough", "nausea"}), "flu"));
-    data.add(new BasicDatum<String, String>(Arrays.asList(new String[]{"cough", "congestion"}), "cold"));
+  private Dataset<String, String> data = new Dataset<>();
 
-    // data.summaryStatistics();
+  @Before
+  public void DSsetUp() {
```
Comment (julwil) [inline test/src/edu/stanford/nlp/classify/DatasetTest.java:19]: @RinorSefa 
we should change this signature to camelcase: `dsSetUp()`

**Review by julwil — COMMENTED** [review]
@RinorSefa 
Looks good to me :)

**Review by jvdassen — APPROVED** [review]
Looks fine!
I will merge & close it since Julius also expressed his approval in the latest comment

**Pull request merged**